Adds new security laser sprites and flavor examine descriptions

Stokes52 opened the pull request:
## What Does This PR Do
Adds new item and in-hand sprites for the laser gun and energy.
Adds new flavorful descriptions when examined, for the LORE!

![guns_comparison](https://user-images.githubusercontent.com/73205477/114356003-f535be00-9b24-11eb-91e8-2a560cd10d16.png)

I've tried to strike a balance between being similar to the old silhouette for gameplay/familiarity while also giving a sleek and updated look.

The new flavor descriptions are just for fun. One references the (intentional) design similarities between the laser and the WT-550 security rifle, and the other references an existing in-game corporation, because more lore is always great.

## Why It's Good For The Game
Updates the very bland looking sprites of the two most common weapons and adds interesting flavor text to make using these weapons feel more immersive.  References existing lore and adds some lore.

## Images of changes
Laser
![laser_description](https://user-images.githubusercontent.com/73205477/114355862-d20b0e80-9b24-11eb-9fb3-76d5d9a7d44d.png)
![laser_gun](https://user-images.githubusercontent.com/73205477/114355878-d505ff00-9b24-11eb-957c-d03716cace9d.png)
![laser_hand](https://user-images.githubusercontent.com/73205477/114355904-db947680-9b24-11eb-9a71-2a660a47edba.gif)

Energy
![energy_description](https://user-images.githubusercontent.com/73205477/114355946-e64f0b80-9b24-11eb-8161-63cd4ccf18bd.png)
![energy_gun](https://user-images.githubusercontent.com/73205477/114355958-e8b16580-9b24-11eb-86d5-66b28860873f.png)
![energy_hand](https://user-images.githubusercontent.com/73205477/114355968-ea7b2900-9b24-11eb-93c4-4f3e5afe89f0.gif)

Animated!
![guns_empty](https://user-images.githubusercontent.com/73205477/114356029-fd8df900-9b24-11eb-93a0-a0dfcf3f479a.gif)

![guns_full](https://user-images.githubusercontent.com/73205477/114356040-0088e980-9b25-11eb-9b79-4796f9c95789.gif)


## Changelog
:cl:
add: New laser gun sprite + in hand sprite
add: New energy gun sprite + in hand sprite
add: New laser gun flavor description
add: New energy gun flavor description
fix: The tesla revolver ammo sprite was broken. I changed a 1 to a 0 so it appears again
/:cl:

dearmochi commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -10,6 +10,7 @@
 	ammo_x_offset = 3
 	flight_x_offset = 15
 	flight_y_offset = 10
+	shaded_charge = 1
```
```suggestion
	shaded_charge = TRUE
```

dearmochi commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -30,6 +31,7 @@
 	charge_sections = 3
 	can_flashlight = 0 // Can't attach or detach the flashlight, and override it's icon update
 	actions_types = list(/datum/action/item_action/toggle_gunlight)
+	shaded_charge = 0
```
```suggestion
```

The `mini` sprite has charge states as well.

dearmochi commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -51,6 +53,7 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/electrode/hos, /obj/item/ammo_casing/energy/laser/hos, /obj/item/ammo_casing/energy/disabler)
 	ammo_x_offset = 4
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	shaded_charge = 0
```
```suggestion
```

Ditto

dearmochi commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -79,6 +82,7 @@
 	can_flashlight = 0
 	trigger_guard = TRIGGER_GUARD_NONE
 	ammo_x_offset = 2
+	shaded_charge = 0
```
```suggestion
```

Same

dearmochi commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -92,3 +96,4 @@
 	ammo_x_offset = 1
 	ammo_type = list(/obj/item/ammo_casing/energy/electrode, /obj/item/ammo_casing/energy/laser, /obj/item/ammo_casing/energy/disabler)
 	selfcharge = 1
+	shaded_charge = 0
```
```suggestion
```

Same

dearmochi commented on code/modules/projectiles/guns/energy/stun.dm:
```diff
@@ -15,7 +15,7 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/shock_revolver)
 	can_flashlight = 0
-	shaded_charge = 1
+	shaded_charge = 0
```
```suggestion
```

dearmochi commented on code/modules/projectiles/guns/energy/stun.dm:
```diff
@@ -24,6 +24,7 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/electrode, /obj/item/ammo_casing/energy/disabler)
 	origin_tech = "combat=4"
 	ammo_x_offset = 2
+	shaded_charge = 0
```
```suggestion
```

dearmochi reviewed (commented):
A few things, also is `laser` supposed to be constantly blinking?

Stokes52 commented:
> A few things, also is `laser` supposed to be constantly blinking?

It follows the design of the "wt550" aka security auto rifle which also blinks.   HOWEVER, if that's too distracting, it can be removed to not blink. I was torn on it too.

@dearmochi 
Regarding the additions of shaded_charge = 0.  Without that added to the other guns, their ammo sprites will break.  Since I modified the energy gun to be shaded_charge = 0 (or shaded_charge = TRUE), it will break anything that is using the energy gun as a base.

The tesla revolver was one I discovered to be broken during my testing as well so I fixed it.

S34NW commented:
These look superb, good work

E-MonaRhg commented:
Okay so first of all, amazing sprites, these all look great, good work. But I got _some points_ to make.

1. It might just be me, but the way the energy gun power indicator cuts the middle "block" when it is at 50% power looks a bit meh, something like this (behold, my non-existent spriting abilities) would _maybe_ look better.
![image](https://user-images.githubusercontent.com/71735193/114382442-a754a080-9b8c-11eb-9936-81acc8db3981.png)
2. The "golden shield of Nanotrasen Security" thing MIGHT be incorrect as the "sliver-kite shield with an engraved sword—of the Nanotrasen security forces" is what is on the CC officer berets, unless you are making a new, _normal security shield_ then fair enough.
![image](https://user-images.githubusercontent.com/71735193/114383019-598c6800-9b8d-11eb-81b9-cb757522e793.png)

3. This one is really minor because lore but, I don't think Warp-Tac Inc. exists within ours, but if you're making your own then more power to you and this point will be irrelevant. And Shellguard Munitions apparently specialises in anti-synthetic weapon systems and platforms, but I mean they still can make normal energy guns I guess, so once again irrelevant.

LightFire53 commented:
> 2. The "golden shield of Nanotrasen Security" thing MIGHT be incorrect as the "sliver-kite shield with an engraved sword—of the Nanotrasen security forces" is what is on the CC officer berets, unless you are making a new, _normal security shield_ then fair enough.

I'm pretty sure that if you inspect the sec uniforms (and especially the corporate security uniforms), it references the golden shield. CC security would be different of course because at that point you are an elite officer, not just a mall cop.

E-MonaRhg commented:
> I'm pretty sure that if you inspect the sec uniforms (and especially the corporate security uniforms), it references the golden shield. CC security would be different of course because at that point you are an elite officer, not just a mall cop.

All security jumpsuits have some flavor of 
`"It's made of a slightly sturdier material than standard jumpsuits, to allow for robust protection."`

And the beret has
`"A beret with the security insignia emblazoned on it. For officers that are more inclined towards style than safety."`

Closest to would be the security jacket
`"A sturdy black jacket with reinforced fabric. Bears insignia of NT corporate security."`

I don't see any reference to "the golden shield" anywhere, but hey, maybe its a good moment to start having it.

Stokes52 commented:
At the moment the golden shield is meant to match the insignias/color scheme found on the corporate security hat, the HOS beret, and the security jacket.  Generally speaking gold is the accent color of choice for security which is why I chose to go that route. 

![gold](https://user-images.githubusercontent.com/73205477/114420359-191af300-9b69-11eb-8080-2dbb2cde34f6.png)

The main point of the blue/gold is functional however. No other weapon has the golden shield making it immediately distinguishable from its peers.

Stokes52 commented:
After thinking about it, I decided to remove the green flashing light so that they both have consistent behavior:
Green = good
Red blinking = no ammo.

I also fixed the flashlight sprite offset so it sits on the railing properly.

![update1](https://user-images.githubusercontent.com/73205477/114654796-bed07e00-9c9f-11eb-96b1-6d6d280f5c2a.gif)

This PR should be ready to go.

McRamon commented:
Overall shape is good, however grip on laser gun just screams "ballistics" while the gun is energy (i don't think energy gun has any recoil). Which is not good, gun doesn't feel like cheap laser rifle anymore. A bit better in terms of quality than current, but not great either.

Stokes52 commented:
Ergonomics is still important for aim and comfort, regardless of recoil. :)

Necaladun reviewed (approved):
@McRamon 's comments make sense, I'd like them to have more of a "cheap mass produced" feel because NT are cheap, however these are so much nicer than the others still I'd approve them as is, but feel free to tweak and re-poke me.
